internal/app/accrual/client.go

```diff
@@ -0,0 +1,117 @@
+package accrual
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"go-developer-course-diploma/internal/app/configs"
+	"go-developer-course-diploma/internal/app/model"
+	"go-developer-course-diploma/internal/app/storage/repository"
+	"net/http"
+	"time"
+)
+
+const (
+	New            = "NEW"
+	Processing     = "PROCESSING"
+	Invalid        = "INVALID"
+	Processed      = "PROCESSED"
+	pollingTimeout = 1 * time.Second
+)
+
+type Client struct {
+	Config                *configs.Config
+	Logger                *logrus.Logger
+	OrderRepository       repository.OrderRepository
+	TransactionRepository repository.TransactionRepository
+}
+
+func NewAccrualClient(cfg *configs.Config, logger *logrus.Logger, orderStore repository.OrderRepository, transactionStore repository.TransactionRepository) *Client {
+	return &Client{
+		Config:                cfg,
+		Logger:                logger,
+		OrderRepository:       orderStore,
+		TransactionRepository: transactionStore,
+	}
+}
+
+func (c *Client) UpdatePendingOrders(orders []string) error {
+	c.Logger.Debug("UpdatePendingOrders: start")
+	for _, o := range orders {
+		link := fmt.Sprintf("%s/api/orders/%s", c.Config.AccrualSystemAddress, o)
+		req, err := http.NewRequest(http.MethodGet, link, nil)
+		if err != nil {
+			return err
+		}
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		var order *model.Order
+		if err := json.NewDecoder(resp.Body).Decode(&order); err != nil {
+			return err
+		}
+		defer resp.Body.Close()
```

Comment:
клиентом по сути должен стать этот кусок, чтобы его можно было бы заменить на
```suggestion
		order, err := c.accuralProvider.GetOrder(o)
```

Reply:
Бенефит от клиента в том, что если однажды внутри системы будет запущен accural 2.0 имеющий (допустим) другой протокол взаимодействия, то тебе будет достаточно написано новый клиент и подменить в бизнес логике.